internal/harnesses/k3s/k3s.go

```diff
@@ -199,6 +211,36 @@ KUBECONFIG=/etc/rancher/k3s/k3s.yaml k3s kubectl config set-cluster default --se
 				return fmt.Errorf("creating kubeconfig: %w", err)
 			}
 
+			if h.opt.HostKubeconfigPath != "" {
+				log.Info(ctx, "Writing kubeconfig to host", "path", h.opt.HostKubeconfigPath)
+				kr, err := h.service.Exec(ctx, provider.ExecConfig{
+					Command: `KUBECONFIG=/etc/rancher/k3s/k3s.yaml kubectl config view --raw &2> /dev/null`,
+				})
+				if err != nil {
+					return fmt.Errorf("writing kubeconfig to host: %w", err)
+				}
+
+				data, err := io.ReadAll(kr)
+				if err != nil {
+					return fmt.Errorf("reading kubeconfig from host: %w", err)
+				}
+
+				cfg, err := clientcmd.Load(data)
+				if err != nil {
+					return fmt.Errorf("loading kubeconfig: %w", err)
+				}
+
+				_, ok := cfg.Contexts["default"]
```

Comment:
Why is "default" significant here?

Reply:
it maps to the `clusters[]`:

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: ...
    server: https://127.0.0.1:41446
  name: default
contexts:
- context:
    cluster: default
    user: default
  name: default
current-context: default
kind: Config
preferences: {}
users:
- name: default
  user:
    client-certificate-data: ...
    client-key-data: ...
```

although now that I think of it, it should just check for `clusters[]` directly... fix inc!